library/ajax/upload.php

```diff
@@ -51,18 +63,62 @@
 $patient_id = filter_input(INPUT_GET, 'patient_id');
 $category_id = filter_input(INPUT_GET, 'parent_id');
 
+if ($isPortal ?? false) {
+    $owner = $GLOBALS['userauthorized'];
+    $files = getMultiple();
+    if (count($files["file"] ?? []) > 0) {
+        $messageService = new MessageService();
+        $data = [];
+        $note['groupname'] = 'Default';
+        // will send to all auth'ed portal users
+        $note['to'] = 'portal-user';
+        $note['from'] = 'portal-user';
+        $note['message_status'] = 'New';
+        $note['title'] = 'New Document';
+        foreach ($files["file"] as $file) {
+            $name = $file['name'];
+            $type = $file['type'];
+            $tmp_name = $file['tmp_name'];
+            $size = $file['size'];
+            $data = addNewDocument(
+                $name,
+                $type,
+                $tmp_name,
+                '',
+                $size,
+                $owner,
+                $patient_id,
```

Comment:
should this instead be $pid (since this is set by the session above) rather than $patient_id (which is taken from a get parameter)?